src/github/api/client.rs

```diff
@@ -164,101 +200,116 @@ impl GithubRepositoryClient {
         branch: &str,
         sha: &CommitSha,
     ) -> anyhow::Result<Vec<CheckSuite>> {
-        measure_network_request("get_check_suites_for_commit", || async {
-            #[derive(serde::Deserialize, Debug)]
-            struct CheckSuitePayload {
-                conclusion: Option<String>,
-                head_branch: String,
-            }
+        perform_network_request(
+            "get_check_suites_for_commit",
+            || async {
+                #[derive(serde::Deserialize, Debug)]
```

Comment:
Let's move these struct definitions outside the closure, to avoid the unnecessary indentation.